tests/Infrastructure.Tests/Favorites/FavoriteRepositoryTests.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Linq;
+using Application.Contracts;
+using Application.FavoriteContracts;
+using Application.Users;
+using Domain.Contracts;
+using Domain.Users;
+
+using Infrastructure.Databases;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Infrastructure.Tests.Contracts;
+
+[Collection("DatabaseTests")]
+public class FavoriteRepositoryTests : IClassFixture<TestDatabaseFixture>
+{
+    private readonly IFavoriteContractRepository _cut;
+    private EFDatabaseContext _context;
+
+    public FavoriteRepositoryTests(TestDatabaseFixture fixture)
+    {
+        _context = fixture.CreateContext();
+        _cut = new EFFavoriteRepository(_context, Mock.Of<ILogger<EFFavoriteRepository>>());
+    }
+
+    [Fact]
+    public void Add_MarksContractAsFavorite_WhenUserAndContractExist()
+    {
+        // Arrange
+        User user = new() { Name = "user" };
+        Contract contract = new();
+
+        _context.Users.Add(user);
+        _context.Contracts.Add(contract);
+        _context.SaveChanges();
+
+        // Act
+        _cut.Add(user.Name, contract.Id);
+
+        // Assert
+        var contracts = _context.Users.Where(u => u.Id == user.Id).Include(c => c.Contracts);
+        contracts.Where(c => c.Id == contract.Id).Should().NotBeNull();
```

Comment:
nit:
What you're fetching here are not the contracts, it's a list of users that *include* the contracts. Because of this you've incorrectly asserted. The assertion still succeeds because you're asserting that `.Where(....).Should().NotBeNull()` but `.Where()` returns an empty collection when it doesn't match on anything, so since an empty collection isn't null then the test still passes (even though it shouldn't).

```suggestion
        var databaseUser = _context.Users.Find(user.id).Include(u => u.Contracts);
        databaseUser.Contracts.Should().Contain(contract);
```